specification/protocol/design-goals.md

```diff
@@ -0,0 +1,19 @@
+# Design Goals for OpenTelemetry Wire Protocol
```

Comment:
Can we put these documents in `/specification` instead of `/specification/protocol`?

Reply:
Yes, I we can move after we merge this PR. I wanted to keep the patch exactly as it was produced by the git commands I used for others to verify.

There is a few other cleanup steps needed after the PR is merged (update links, add link checking, etc).